shared/src/logging.rs

```diff
@@ -0,0 +1,57 @@
+use std::str::FromStr;
+
+#[derive(Clone, Copy)]
+pub enum LogFmt {
+    Plain,
+
+    #[cfg(feature = "gcp")]
+    Gcp,
+}
+
+impl FromStr for LogFmt {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "plain" => Ok(LogFmt::Plain),
+
+            #[cfg(feature = "gcp")]
+            "gcp" => Ok(LogFmt::Gcp),
+
+            _ => Err("Unrecognized log format"),
+        }
+    }
+}
+
+fn set_up_plain_log_format(_log_format: LogFmt) {
+    tracing_subscriber::fmt()
+        .with_env_filter(
+            tracing_subscriber::EnvFilter::try_from_default_env()
+                .unwrap_or(tracing_subscriber::EnvFilter::new("info")),
+        )
+        .init();
+}
+
+#[cfg(not(feature = "gcp"))]
+pub fn init_logger(log_format: LogFmt) {
+    set_up_plain_log_format(log_format)
+}
+
+#[cfg(feature = "gcp")]
+pub fn init_logger(log_format: LogFmt) {
+    match log_format {
+        LogFmt::Plain => set_up_plain_log_format(log_format),
```

Comment:
If we move the `gcp` feature down here, we  don't need the above two functions

Reply:
Good point 👍

Reply:
the body of `set_up_plain_log_format` can be moved here too now, as it's not used anywhere else :+1: